### PR TITLE
Basic code to deploy Azure gateways

### PR DESCRIPTION
**Pull request by mkolesnik**
This is a basic deployer of Azure gateway using MachineSet, it will be
further extended.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: pkg/azure/gw-machineset.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+package azure
+
+var machineSetYAML = `apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: {{.InfraID}}
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: {{.InfraID}}-submariner-gw-{{.Region}}{{.AZ}}
```
Comment (Jaanki) [inline pkg/azure/gw-machineset.go:27]: Do we want to have such long name?

File: pkg/azure/gw-machineset.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            resourceID: >-
+              /resourceGroups/{{.InfraID}}-rg/providers/Microsoft.Compute/images/{{.InfraID}}
+            sku: ""
+            version: ""
+          internalLoadBalancer: ""
+          kind: AzureMachineProviderSpec
+          location: {{.Region}}
+          managedIdentity: {{.InfraID}}-identity
+          natRule: null
+          networkResourceGroup: {{.InfraID}}-rg
```
Comment (Jaanki) [inline pkg/azure/gw-machineset.go:67]:
```suggestion
          networkResourceGroup: {{.BaseGroupName}}
```
BaseGroupName is part of azureCloud https://github.com/submariner-io/cloud-prepare/pull/203/files#diff-f2027c3046b53bdf075254bbf73d790797640745c6d58bff13b71992570287feR38

Comment (Jaanki) [reply pkg/azure/gw-machineset.go:67]: Applies everywhere.

File: pkg/azure/gw-machineset.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package azure
+
+var machineSetYAML = `apiVersion: machine.openshift.io/v1beta1
```
Comment (Jaanki) [inline pkg/azure/gw-machineset.go:20]: you can use `go:embed` directive and then read it as string/file. See for eg  https://github.com/submariner-io/submariner-operator/blob/devel/pkg/subctl/cmd/gather/summary.go#L40

**Review by Jaanki — CHANGES_REQUESTED** [review]

**Comment by dfarrell07** [comment]
@aswinsuryan will work on this, but will likely create a fresh PR. Closing for now.

**Pull request closed without merging**